src/FRWeights.cc

```diff
@@ -0,0 +1,49 @@
+#include "UserCode/llvv_fwk/interface/FRWeights.h"
+#include <TFile.h>
+#include <TSystem.h>
+#include <math.h>
+#include <vector>
+#include <iostream>
+
+using namespace std;
+
+/*****************************************************************/
+FRWeights::FRWeights()
+/*****************************************************************/
+{
+}
+
+/*****************************************************************/
+FRWeights::~FRWeights()
+/*****************************************************************/
+{
+}
+
+/*****************************************************************/
+bool FRWeights::init(const string& WeightsFileName)
+/*****************************************************************/
+{
+  WeightsFile = TFile::Open(WeightsFileName.c_str());
+  if(!WeightsFile) {
+    cout<<"ERROR: Cannot open weights file "<<WeightsFileName<<"\n";
+    return false;
+  } else {
+    return true;
+  } 
```

Comment:
cache everything and close the file when done
otherwise we will have huge trouble with file access